nixos/modules/services/desktops/espanso.nix

```diff
@@ -25,10 +25,33 @@ in
   };
 
   config = lib.mkIf cfg.enable {
+    security.wrappers.espanso = lib.mkIf (cfg.package == pkgs.espanso-wayland) {
+      capabilities = "cap_dac_override+p";
+      owner = "root";
+      group = "root";
+      source = lib.getExe (
+        pkgs.espanso-wayland.overrideAttrs (old: {
+          patchPhase =
+            old.patchPhase or ""
+            + ''
+              substituteInPlace espanso/src/cli/daemon/mod.rs \
```

Comment:
This should also be moved inside the package definition, behind a flag (I lack the creativity for a proper name, but something like `inSecurityWrapper`). Also, use `postPatch` instead of `patchPhase` for such substitutions.

Reply:
Even better, the flag could be the path to the exe, e.g.

```nix
{ ..., securityWrapperPath ? null }:
stdenv.mkDerivation {
  ...

  postPatch = lib.optionalString securityWrapperPath != "" ''
    substituteInPlace espanso/src/cli/daemon/mod.rs \
      --replace-fail \
        'std::env::current_exe().expect("unable to obtain espanso executable location");' \
        'std::ffi::OsString::from("${securityWrapperPath}");'
  '';

  ...
}

Reply:
Although I suppose the `securityWrapperPath` should be assumed to be `null` in those cases, so it shouldn't affect them...

Reply:
Yes, my suggestion is mainly meant to make it easier to fix the patch on updates, having it all in one place.